rllib/agents/trainer.py

```diff
@@ -747,34 +771,43 @@ def env_creator_from_classpath(env_context):
             self.evaluation_workers = self._make_workers(
                 env_creator=self.env_creator,
                 validate_env=None,
+                # TODO: Replace with self.get_default_policy_class(),
                 policy_class=self._policy_class,
                 config=evaluation_config,
                 num_workers=self.config["evaluation_num_workers"])
 
-    @DeveloperAPI
+    # TODO: Deprecated: In your sub-classes of Trainer, override `setup()`
+    #  directly and call super().setup() from within it if you would like the
+    #  default setup behavior.
     def _init(self, config: TrainerConfigDict,
               env_creator: Callable[[EnvContext], EnvType]) -> None:
-        """Subclasses should override this for custom initialization.
+        pass
 
-        In the case of Trainer, this is called from inside `self.setup()`.
+    @ExperimentalAPI
+    def get_default_policy_class(self, config: PartialTrainerConfigDict):
+        """Returns a default Policy class to use, given a config.
 
-        Args:
-            config: Algorithm-specific configuration dict.
-            env_creator: A callable taking an EnvContext as only arg and
-                returning an environment (of any type: e.g. gym.Env, RLlib
-                BaseEnv, MultiAgentEnv, etc..).
+        This class will be used inside RolloutWorkers' PolicyMaps in case
+        the policy class is not provided by the user in any single- or
+        multi-agent PolicySpec.
+
+        This method is experimental and currently only used, iff the Trainer
+        class was not created using the `build_trainer` utility and if
+        the Trainer sub-class does not override `_init()` and create it's
+        own WorkerSet in `_init()`.
         """
         raise NotImplementedError
 
     @override(Trainable)
-    @PublicAPI
-    def train(self) -> ResultDict:
-        """Overrides super.train to synchronize global vars."""
+    def step(self):
```

Comment:
We don't override `Trainable.train()` anymore (which was never a good idea anyways as sub-classes of `Trainable`  should only override `setup()`.